1a.cpp

```diff
@@ -4,23 +4,24 @@ using namespace std;
 
 int main()
 {
-	vector<int> v1(5);
+	cout<<"Введите размер вектора\n";
+	int vecsize;
+	cin>> vecsize;
+	vector<int> v1(vecsize);
 	cout <<"Размер вектора: " <<v1.size()<<"\n";
 	cout <<"Коллекция чисел: \n";
-	int i;
-	for(i=4; i>=0; i--)
+	int counter=0;
+	for(int i=0; i<vecsize; i++)
```

Comment:
Use `std::vector::size` method here and use it whenever possible, it will always return the correct size, so you don't have to worry about your local variable to be in synch with the container.